File: MGWDev.PnPFramework.UT/MGWDev.PnPSDK.Lib/PnPListItemProvider.cs
using MGWDev.PnP;
using MGWDev.PnP.Model;
using PnP.Core.Model.SharePoint;
using PnP.Core.Services;
using System;
using System.Collections.Generic;

namespace MGWDev.PnPSDK.Lib
{
    public class PnPListItemProvider : IListItemProvider<MyTestListItem>
    {
        protected PnPContext Context { get; set; }
        protected string ListTitle { get; set; } = "DevList";
        public PnPListItemProvider(PnPContext context)
        {
            Context = context;
        }
        public List<MyTestListItem> GetMyItems()
        {
            IList list = Context.Web.Lists.GetByTitle(ListTitle);
            IListItemCollection collection = list.Items;

            List<MyTestListItem> result = new List<MyTestListItem>();
            foreach(IListItem item in collection)
            {
                result.Add(new MyTestListItem()
                {
                    Id = item.Id,
                    Modified = (DateTime)item["Modified"],
                    Title = item.Title
                });
            }

            return result;
        }

        public void UpdateItem(MyTestListItem item)
        {
            throw new NotImplementedException();
        }
    }
}

File: spfx-webpart/src/dal/PnPListItemProvider.ts
import { SPFI } from "@pnp/sp";
import { ISPListItem } from "../model/ISPListItem";
import { IListItemProvider } from "./IListItemProvider";
import "@pnp/sp/webs";
import "@pnp/sp/lists";
import "@pnp/sp/items";

export class PnPListItemProvider<T extends ISPListItem> implements IListItemProvider<T>{
    
    constructor(protected spfi: SPFI, protected listName: string) { }
    
    public async getListItems(): Promise<T[]> {
        return await this.spfi.web.lists.getByTitle(this.listName).items();
    }
    public async getById(id: number): Promise<T> {
        return await this.spfi.web.lists.getByTitle(this.listName).items.getById(id)();
    }
}